til-api/src/main/java/com/til/controller/problem/reponse/ProblemListResponse.java

```diff
@@ -0,0 +1,14 @@
+package com.til.controller.problem.reponse;
+
+import java.util.List;
+
+import com.til.domain.problem.dto.ProblemInfoDto;
+
+public record ProblemListResponse(
+                                  List<ProblemInfoDto> problemInfoDtoList
```

Comment:
P3: 변수명이 현재 API response 반환될 때 묶이는 리소스명이어서 dto 같은 네이밍은 빠져도 괜찮을 것 같습니다!
```suggestion
                                  List<ProblemInfoDto> problemList
```

Reply:
ProblemDTO는 상세화면에서도 같이 사용될 부분이여서 컬럼을 변경하기보다
dto안에 아래와 같이 메소드를 추가하는 방법으로 구현하면 될까요?
```java 
public static ProblemInfoDto forList(Problem problem) {
        return ProblemInfoDto.builder()
            .title(problem.getTitle())
            .score(problem.getScore())
            .build();
    }
```
아니면 아예 상세화면을 위한 dto와 리스트를 위한 dto를 따로 구현하는게 나을까요?

Reply:
분리하는게 좋을 수도 있을 것 같습니다!  
상세 화면에서는 다른 부분에서 데이터를 엮어서 리스트와는 형태가 다른 데이터를 가져올 수도 있을 것 같아서 관심사를 분리해도 좋을 것 같습니다😀